packages/firestore/src/local/simple_query_engine.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * @license
+ * Copyright 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { SnapshotVersion } from '../core/snapshot_version';
+import { Query } from '../core/query';
+import { DocumentMap } from '../model/collections';
+import { assert } from '../util/assert';
+import { LocalDocumentsView } from './local_documents_view';
+import { QueryEngine } from './query_engine';
+import { PersistenceTransaction } from './persistence';
+import { PersistencePromise } from './persistence_promise';
+
+/**
+ * A naive implementation of QueryEngine that just loads all the documents in
+ * the queried collection and then filters them in memory.
+ */
+export class SimpleQueryEngine implements QueryEngine {
+  private localDocumentsView: LocalDocumentsView | undefined;
+
+  setLocalDocumentsView(localDocuments: LocalDocumentsView): void {
+    this.localDocumentsView = localDocuments;
+  }
+
+  /** Returns all local documents matching the specified query. */
+  getDocumentsMatchingQuery(
+    transaction: PersistenceTransaction,
+    query: Query,
+    sinceReadTime: SnapshotVersion
+  ): PersistencePromise<DocumentMap> {
+    assert(
+      this.localDocumentsView !== undefined,
+      'setLocalDocumentsView() not called'
+    );
+
+    // TODO: Once LocalDocumentsView provides a getCollectionDocuments()
+    // method, we should call that here and then filter the results.
```

Comment:
Hrm.  What's the context for this TODO (other than the fact that it exists in Android too)?  Why do we expect LocalDocumentsView is going to provide a `getCollectionDocuments()` method in the future?

Reply:
Hm. I don't know. I ported it cause it was there an Android. Unfortunately, all traces of who added it are gone by now. I could invest some research and dig this up on Codesearch, but I suspect you would not like the outcome :)

Reply:
Ohhh...  GRRRR... I forgot about that.  I suspect it used to be TODO(indexing) which would imply it doesn't apply to web.  So I would prefer if you don't port it.  But that means you have to run CI again, so maybe not worth removing (it's friday, YOLO, etc.)

Reply:
I'll drop it in the next PR.